datafusion/proto/src/bytes/mod.rs

```diff
@@ -93,6 +101,69 @@ impl Serializeable for Expr {
     }
 }
 
+/// Serialize a LogicalPlan as bytes
```

Comment:
I wonder if it is time to think about a slightly more encapsulated API, something that would allow

```rust
  let bytes = PlanSerializer::new()
    .with_extension_codec(&extension_codec)
   .serialize()?;
```

Rather than having two separate free functions 🤷

Reply:
Yes, I had been thinking about this too and we will need a unified API to implement subquery support (because expressions now reference plans) so I think we can make this API change as part of https://github.com/apache/arrow-datafusion/issues/2640